lib/urldata.h

```diff
@@ -1461,6 +1461,10 @@ struct UrlState {
   trailers_state trailers_state; /* whether we are sending trailers
                                     and what stage are we at */
 #endif
+#ifndef CURL_DISABLE_PROXY
+  /* to keep track whether we already sent PROXY header or not */
+  BIT(is_proxy_hdr_sent);
```

Comment:
Do I need to initialise the state value explicitly?

Reply:
No, but maybe you could call it "haproxy" instead of just proxy to make it abundantly clear what kind of proxy it is used for?

Reply:
Done.